docs/HowToFAQ/FAQLoopModifies.md

```diff
@@ -0,0 +1,58 @@
+---
+title: If I have an assertion about a class and a loop that doesn't mention (read, modify)  the class, why does dafny fail to establish the assertion after the loop?
+---
+
+## Question
+
+If I have an assertion about a class and a loop that doesn't mention (read, modify)  the class, why does dafny fail to establish the assertion after the loop?
+
+## Answer
+
+TODO - This may need adjustment of the modifies/assignmetn targets discussion
+
+The short answer is that you need an appropriate combination of modifies clause and loop invariants to prove assertions about the end result of a loop.
+
+In Dafny's way of reasoning about a loop (which is typical of verifier systems), the loop invariant is the only thing you can assume at the top of every iteration. In other words, at the top of each loop iteration, it is as if all variables have arbitrary values, modulo the loop invariant. Sometimes, the word _havoc_ is used to describe this: the verifier havocs all variables, that is, gives all variables arbitrary values, and then assumes the loop invariant.
```

Comment:
I find this style confusing: first we state something that is wrong, and then we say "not quite"

Reply:
In this case I copied Rustan's answer to the question.

I reread it and found for myself the development of the answer helpful and not confusing.. But if you do think users would find the discussion confusing, I'll do a full rewrite.

Reply:
I'd like to get a second opinion, but it's not worth blocking the whole PR on.  Can we get maybe @MikaelMayer ?